source/stylist/configuration.py

```diff
@@ -27,13 +27,20 @@
 
 
 class Configuration(ABC):
+    """
+    Core configuration functionality.
+    """
     # Pycharm is confused by referring to the class you are defining. mypy is
     # fine with it.
     #
     def __init__(self,
-                 parameters: Mapping[str,
-                                     Mapping[str, str]] = None,
-                 defaults: 'Configuration' = None):
+                 parameters: Optional[Mapping[str,
+                                              Mapping[str, str]]] = None,
+                 defaults: Optional['Configuration'] = None):
+        """
+        :param parameters: Maps section to key/value pairs.
```

Comment:
Yes, I use a sphinx extension which uses the type hinting to infer that information.

I'm not sure how it will render but the generated output of that line looks like this:

```
Parameters
        parameters (Optional[Mapping[str, Mapping[str, str]]]) – Maps section to key/value pairs.
```